File: fixtures/scale/julia/medium/src/Format.jl
module Format

using ..Config

export fmt_result

function fmt_number(value::Float64)
    precision = Config.config_get_precision()
    return string(round(value; digits = precision))
end

function fmt_result(value::Float64)
    return "= " * fmt_number(value)
end

function fmt_hex(value::Int)
    return string("0x", string(value, base = 16))
end

end

File: fixtures/scale/r/medium/core/format.r
# Result formatting helpers.

fmt_number <- function(value) {
  precision <- config_get("precision")
  formatC(value, digits = precision, format = "g")
}

fmt_result <- function(value) {
  paste0("= ", fmt_number(value))
}

fmt_hex <- function(value) {
  sprintf("0x%x", as.integer(value))
}
